web/pgadmin/llm/reports/queries.py

```
@@ -0,0 +1,907 @@
+
+    sql = query_def['sql']
+
+    # Check if query requires an extension
+    required_ext = query_def.get('requires_extension')
+    if required_ext:
+        # Check if extension is installed
+        check_sql = f"""
+            SELECT EXISTS (
+                SELECT 1 FROM pg_extension WHERE extname = '{required_ext}'
+            ) as available
+        """
```

Ruff (0.14.8)

Possible SQL injection vector through string-based query construction

(S608)